### ISSUE TITLE
Feature Request: ResizeMode.Min

### ISSUE THREAD

**Issue by phillip-haydon** [JimBobSquarePants/ImageProcessor#209]
There is currently a `ResizeMode.Max` which takes the max height or width of what the image fits into based on the constraints.

It would be nice to have a `ResizeMode.Min` which does the opposite.

Given an image resize of 1400x800, the resize would scale down until the minimum of those are hit.

i.e if the min is 1600x800 then it takes the height as the min because in order to scale down further would mean the height is less than the min.

Likewise the opposite is true. When the min becomes 1400x900, to scale down the height more means the width is less than the min.

Hope that makes sense.

**Comment by JimBobSquarePants** [comment]
I think so... I'm not fully awake though. Should be pretty simple to implement though.

**Comment by JimBobSquarePants** [comment]
Having a second look I'm not sure how I'm gonna fit it into the way the current calculations. Could you have a look?

**Comment by phillip-haydon** [comment]
The calculations would be rather hard, you would need to work out both the height and width and work out which scenario doesn't match the desired outcome. Which is a little bit of a hassle.

Could you point me in the right direction of where this would sit in the code base and I'll take a look at how we can calculate this. 

Also I'll be on holiday from Thursday for a couple of weeks. So I'll try figure out the best way to calculate it so we can implement when I get back. I have motivation to do this because I need this functionality in order to generate Cover (css) photos for my website.

**Comment by JimBobSquarePants** [comment]
Yeah of course. Here's the method where I work out the correct output size. https://github.com/JimBobSquarePants/ImageProcessor/blob/e62d4ada310d9ebe1450cd52566bf02af970d410/src/ImageProcessor/Imaging/Resizer.cs#L197

**Comment by phillip-haydon** [comment]
Awesome, I'll clone the project and take a look this week on the plane, see if I can come up with something.

**Comment by JimBobSquarePants** [comment]
Ace cheers!

**Comment by JimBobSquarePants** [comment]
Hopefully I catch you before your holidays with this...

Could you give me an indicator of expected values from a resize. I can't quite get my head round what you are after?

e.g 
```
sourceWidth x SourceHeight 
instructedWidth x instructedHeight
outputWidth x outputHeight
```

**Comment by phillip-haydon** [comment]
`.Max` takes the height or width that is equal or greater.

i.e if you have your X/Y as 1000 x 1000, and you have an image that is 3000 x 1500, it will resize it down to 1000x500.

I want to do that opposite of that. 

Given X/Y of 1000 x 1000 with the setting as `.Min` I want it to resize down so I end up with a photo that is 2000x1000 because the shortest side being the height reach the minimum value first without going less than the specified value.

Does that make more sense?

So:

* Input: 3000x1500
* Instructed: 1000x1000
* Output: 2000x1000

I'll write up some test cases to use when I get home tonight.

**Comment by phillip-haydon** [comment]
There's a hard to handle scenario where if an image is already smaller by height or width than the instructed input. It would cause scaling up.

IMO it shouldn't scale up, it shouldn't resize at all. But I'm not sure what the best solution for this scenario is.

**Comment by JimBobSquarePants** [comment]
I've added the functionality to the latest build. Behaviour is as follows and outputs your described results:

1. Look for the shortest difference between source length and intended length in each dimension.
2. Resize to the intended length on the result, the other dimension is resized matching the original image ratio.
3. If equal, resize to both intended dimensions.
4. Upscaling is turned off.

**Comment by phillip-haydon** [comment]
Ah man it's working perfectly! The images for the hero shot on my site have come out MUCH clearer now that the sizes are more to the height/width of the viewable area. 

Tried with about 50 images and found no faults. 

:+1: 

Thank you so much!

**Comment by JimBobSquarePants** [comment]
Excellent news! Glad I nailed it. I should be shipping very soon.

**Comment by JimBobSquarePants** [comment]
Fixed in [v2.3.0](https://www.nuget.org/packages/ImageProcessor/2.3.0)

**Comment by phillip-haydon** [comment]
:+1: thank you so much :D
